### ISSUE TITLE
[BUG] In app camera not working

### ISSUE THREAD

**Issue by W1zardK1ng** [aashishksahu/SafeSpace#64]
I'm not able to take any photos or videos using the camera option built inside the app.
Clicking on the camera shutter button for photos does nothing and for the videos it gives an error saying "internal error, could not save video."
app version: 2.0.0, google play store.
Device: google pixel 6a

**Comment by aashishksahu** [comment]
I tried this in my emulator, it's working fine. maybe reinstall the app, see if that fixes the issue, check permissions and also, please provide a few more details such as the android version, custom rom (if any), steps to reproduce the error, and, if possible, a logcat would be very helpful.

**Comment by IrealiTY** [comment]
Same issue on the pixel 8 pro. Tried reinstalling too

**Comment by aashishksahu** [comment]
Hmm... will need a logcat to check.

**Comment by W1zardK1ng** [comment]
Please find the below screenshot.
attaching the logs too, hope this helps.
[Safe Space log 5e3325376748.txt](https://github.com/user-attachments/files/16091650/Safe.Space.log.5e3325376748.txt)

![Screenshot_20240703-193351](https://github.com/aashishksahu/SafeSpace/assets/110819956/4f4be09f-18f8-40b3-869c-ef4d79323446)

**Comment by aashishksahu** [comment]
Please try again and send me the error log from the Logs option in settings.

**Comment by aashishksahu** [comment]
@W1zardK1ng just saw the logs, thanks for that, it will help in diagnosing the issue.

**Comment by aashishksahu** [comment]
@W1zardK1ng I checked the logs and there wasn't enough info that points to a crash. Please also share the logs from the app settings.

**Comment by W1zardK1ng** [comment]
I tried the new app but I was not able to copy the logs. Long pressing on the text does nothing and three is no option to save the log file.

**Comment by aashishksahu** [comment]
Hmm… let me check that, meanwhile can you please share the screenshot.

**Comment by W1zardK1ng** [comment]
Sorry for the delay, but here is the logs from the app after the error.
[safesapce_logs.txt](https://github.com/user-attachments/files/16241518/safesapce_logs.txt)

**Comment by aashishksahu** [comment]
Hey @W1zardK1ng , can you tell me if you are still facing the camera issue? I've been tracking this issue all across the web and it seems it's a rare issue that happens only with some devices. I haven't been able to reproduce this issue on any of my devices (virtual and real) but this error message seems legit. I am still digging into it, just in case a version update helped you, please let me know.

**Comment by W1zardK1ng** [comment]
Yes, i still have the issue on my Pixel 6a a running GrapheneOS.
But yah i tried the app on an old device running LineageOS and the camera is working fine there.

**Comment by aashishksahu** [comment]
So one thing I observed is that, the people who reported having this issue have GrapheneOS. I wonder what's different?
Either they have some different security restriction or maybe some changes to the API itself?

**Comment by Dekkers2000** [comment]
Hi.. Not working either in my pixel 8 pro with grapheneos..

**Comment by aashishksahu** [comment]
Just checked the issue in graphene os tracker, as per the log the camera is off and this is only happening in graphene os. Unless I can install graphene os on an emulator there’s no way for me to debug this. I’ll check if gos can run on emulator

**Comment by clach04** [comment]
In case this helps, I'm not seeing this on Samsung (Android 10). Camera works fine.
